src/barChart.ts

```diff
@@ -577,5 +613,74 @@ module powerbi.extensibility.visual {
 
             return div;
         }
+
+        private getColorValue(color: Fill|string): string {
+            // Override color settings if in high contrast mode
+            if (this.host.colorPalette.isHighContrast) {
+                return this.host.colorPalette.foreground.value;
+            }
+
+            // If plain string, just return it
+            if (typeof(color) === 'string') {
+                return color
+            }
+            // Otherwise, extract string representation from Fill type object
+            return color.solid.color;
+        }
+
+        private initAverageLine() {
+            this.averageLine = this.svg
+                .append('g')
+                .classed('averageLine',true);
+
+            this.averageLine.append('line')
+                .attr('id','averageLine');
+
+            this.averageLine.append('text')
+                .attr('id','averageLineLabel');
+        }
+
+        private handleAverageLineUpdate(height:number, width:number, yScale: d3.scale.Linear<number,number>) {
+            let average = this.calculateAverage();
+            let fontSize = d3.min([height, width]) * BarChart.Config.xAxisFontMultiplier;
+            let chosenColor = this.getColorValue(this.barChartSettings.averageLine.fill);
+            // If there's no room to place lable above line, place it below
```

Comment:
typo: label

Reply:
Thanks, surprisingly there were many instances of this